src/value_generators/src/DocumentGenerator.cpp

```diff
@@ -1298,13 +1300,22 @@ class ObjectGenerator : public Generator<bsoncxx::document::value> {
           _keyGen{stringGenerator(node["havingKeys"], generatorArgs)},
           _valueGen{
               valueGenerator<false, UniqueAppendable>(node["andValues"], generatorArgs, parsers)},
-          _nTimesGen{intGenerator(extract(node, "withNEntries", "^Object"), generatorArgs)} {}
+          _nTimesGen{intGenerator(extract(node, "withNEntries", "^Object"), generatorArgs)},
+          _allowDuplicateKeys{node["allowDuplicateKeys"].to<bool>()} {}
```

Comment:
The implementation is actually *skip* duplicate keys and I think that's actually much better behavior than retrying until we find a unique key.

With this in mind, consider making the syntax

```yaml
^Object:
  withNEntries: 10
  havingKeys: {^Foo}
  andValues: {^Bar}
  # allowDuplicateKeys: bool
  duplicatedKeys: skipped
```

This allows for additional duplicated keys behaviors to be added in the future.

Reply:
Do you mean that you'd like for the duplicated keys to have three possible behaviours:
1. skipped (equivalent in this patch to `allowDuplicateKeys: false`)
2. inserted (equivalent in this patch to `allowDuplicateKeys: true`)
3. retried (would require switching from bool to enum)

I can change the param from a bool to an enum but only add "skipped" and "inserted" values so that "retried" can be implemented in the future when/if it's actually needed. Sounds good?